pkgs/native_assets_cli/lib/src/api/link_output.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'architecture.dart';
+import 'asset.dart';
+import 'build_output.dart';
+import 'os.dart';
+
+extension type LinkOutput(BuildOutputImpl _buildOutput) {
+  /// Start time for the link of this output.
```

Comment:
I don't believe this should be an extension type.

Maintainers of helper packages will have to write unit tests involving equality checks between manually constructed `LinkOutput`s.
Look at the number of tests using `BuildOutput` in package:native_toolchain_c.
We should likely have similar tests doing things with `LinkConfig` and `LinkOutput` for other helper packages.
It should be easy to construct a `LinkOutput()`.